app/models/episode_import.rb

```diff
@@ -103,7 +113,13 @@ def entry_audio_files(entry)
   end
 
   def create_or_update_episode!
-    self.episode ||= Episode.new(podcast: podcast)
+    lookup = Episode.where(podcast_id: podcast.id).find_by_item_guid(guid)
```

Comment:
didn't need this line to split into a different method so moved it here as recommended. now scoped to the specific podcast as well.

Reply:
+7